client/packages/invoices/src/Prescriptions/DetailView/Toolbar.tsx

```diff
@@ -12,28 +12,58 @@ import {
   Formatter,
   DateUtils,
   useConfirmationModal,
+  useTableStore,
+  useDeleteConfirmation,
 } from '@openmsupply-client/common';
 import { PatientSearchInput } from '@openmsupply-client/system';
 import { usePrescription } from '../api';
 import { ClinicianSearchInput } from '../../../../system/src/Clinician';
-import { usePrescriptionRows } from '../api/hooks/line/usePrescriptionRows';
+import { usePrescriptionLines } from '../api/hooks/usePrescriptionLines';
 
 export const Toolbar: FC = () => {
-  const { id, patient, clinician, prescriptionDate, createdDatetime, update } =
-    usePrescription.document.fields([
-      'id',
-      'patient',
-      'clinician',
-      'prescriptionDate',
-      'createdDatetime',
-    ]);
-  const onDelete = usePrescription.line.deleteSelected();
-  const onDeleteAll = usePrescription.line.deleteAll();
-  const { items } = usePrescriptionRows();
+  const {
+    query: { data },
+    update: { update },
+    isDisabled,
+    rows: items,
+  } = usePrescription();
+  const { id, patient, clinician, prescriptionDate, createdDatetime } =
+    data ?? {};
+
+  const selectedRows =
+    useTableStore(state => {
+      return items
+        ?.filter(({ id }) => state.rowState[id]?.isSelected)
+        .map(({ lines }) => lines.flat())
+        .flat();
+    }) || [];
+
+  const {
+    delete: { deleteLines },
+  } = usePrescriptionLines();
+
+  const deleteAll = () => {
+    const allRows = (items ?? []).map(({ lines }) => lines.flat()).flat() ?? [];
+    if (allRows.length === 0) return;
+    deleteLines(allRows);
+  };
 
-  const isDisabled = usePrescription.utils.isDisabled();
   const t = useTranslation();
 
+  const confirmAndDelete = useDeleteConfirmation({
+    selectedRows,
+    deleteAction: () => deleteLines(selectedRows),
+    canDelete: !isDisabled,
+    messages: {
+      confirmMessage: t('messages.confirm-delete-lines', {
+        count: selectedRows.length,
+      }),
+      deleteSuccess: t('messages.deleted-lines', {
+        count: selectedRows.length,
+      }),
+    },
+  });
```

Comment:
This was in a hook before, but I think it makes more sense to have UI stuff in the UI components and only have logic and database interactions in hooks.